### DISCUSSION TITLE
tree-shaking: is it still the default behavior of @babel/preset-env to transform es6 to cjs?

### DISCUSSION THREAD

**Discussion by berkant** [webpack/webpack#12950] [Q&A]
Hi.

Quoting from the https://webpack.js.org/guides/tree-shaking/#conclusion,
> Ensure no compilers transform your ES2015 module syntax into CommonJS modules (**this is the default behavior of the popular Babel preset @babel/preset-env** - see the documentation for more details).

However https://babeljs.io/docs/en/babel-preset-env#modules says the default mod is "auto", where the actual value is determined from the `caller`.

Just wanting to confirm, does docs need a change regarding this?

**Comment by alexander-akait** [comment]
Yep, feel free to update docs

**Comment by nyngwang** [comment]
> @berkant:
> Quoting from the https://webpack.js.org/guides/tree-shaking/#conclusion,
> > [...]
> 
> [...]
> Just wanting to confirm, does docs need a change regarding this?

The documentation is no wrong. This is my test for you:

`babel.config.js`:

```js
presets: [
  [
    '@babel/preset-env', {
      debug: true,
      useBuiltIns: 'usage',
      corejs: 3.21,
      // no `modules: false` here
    },
  ],
// ...
```

`package.json`: (I tried to make the target browser as old as possible so that (_I think_, you can ask @alexander-akait for checking) it should try to transpile to es5 instead of es6.)

```js
"sideEffects": [
  "*.css"
],
"browserslist": [
  "Edge 12"
],
```

Finally, `webpack.config.js`.`optimization`: (I commented out all tree-shaking related hint, but I did add `--mode production` to `webpack` CLI-command script)

```js
// usedExports: true,
// sideEffects: true,
// providedExports: true,
minimize: true,
```

In the result bundle the unused `export function square(x) { ... }`(I used the example from the webpack's docs.) did get removed.

**Reply by JiangWeixian** [reply]
thx, project package.json sideEffects field works for me.
